### PR TITLE
Fix 'load_asset'

### PR DESCRIPTION
**Pull request by ozkriff**

### CODE REVIEW

File: cargo-apk/src/build.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use std::collections::HashSet;
 use std::env;
 use std::fs;
 use std::fs::File;
+use std::os::unix::fs::symlink;
```
Comment (tomaka) [inline cargo-apk/src/build.rs]: That change would break `cargo-apk` on Windows.

**Comment by ozkriff** [comment]
This should fix windows build.

**Comment by tomaka** [comment]
Uhh. You should fix it by using one of the cross-platform functions!

**Comment by ozkriff** [comment]
`hardlink` always says `Operation not permitted` when used with directories, `symlink` is not cross-platform - i can't find other functions :( .

Hmm, maybe it's easier to manually copy files using `walkdir` crate?

**Comment by ozkriff** [comment]
... or i can use `std::os::windows::fs::symlink_dir` in

```rust
#[cfg(target_os = "windows")]
fn build_assets(path: &Path, config: &Config) {...
```

**Comment by tomaka** [comment]
I suggestion doing like this: https://github.com/tomaka/vulkano/blob/master/glsl-to-spirv/build/build.rs#L26-L28

If `hard_link` returns an error, copy instead. I don't think you need to use `walk_dir`, copying the directory itself should work I guess.

**Comment by ozkriff** [comment]
> ... I don't think you need to use walk_dir, copying the directory itself should work I guess.

`the source path is not an existing regular file`

https://github.com/rust-lang/rust/blob/992bb1332/src/libstd/fs.rs#L1959-L1968

**Comment by Thraein** [comment]
I tried a few things, copying only the directory definitely doesn't work and using a recursive directory copy is better suited for a build script.

I've got a working solution with delegating to the os specifc functions (only tested on windows yet):

```rust
#[cfg(target_os = "windows")]
fn create_dir_symlink(src_path: &Path, dst_path: &Path) -> Result<()> {
    os::windows::fs::symlink_dir(&src_path, &dst_path)
}

#[cfg(not(target_os = "windows"))]
fn create_dir_symlink(src_path: &Path, dst_path: &Path) -> Result<()> {
    os::unix::fs::symlink(&src_path, &dst_path)
}
```

**Comment by ozkriff** [comment]
@Thraein thanks :)

@tomaka ping? :)

**Pull request merged**